src/Microsoft.AspNetCore.Mvc.ViewFeatures/HtmlHelperOfT.cs

```diff
@@ -370,14 +363,20 @@ public string NameFor<TResult>(Expression<Func<TModel, TResult>> expression)
                 htmlAttributes);
         }
 
-        protected string GetExpressionName<TResult>(Expression<Func<TModel, TResult>> expression)
+        /// <summary>
+        /// Gets the name for the lambda expression.
+        /// </summary>
+        /// <typeparam name="TResult">The type of the result.</typeparam>
+        /// <param name="expression">The expression.</param>
+        /// <returns>The expression name.</returns>
+        public string GetExpressionName<TResult>(Expression<Func<TModel, TResult>> expression)
         {
             if (expression == null)
             {
                 throw new ArgumentNullException(nameof(expression));
             }
 
-            return ExpressionHelper.GetExpressionText(expression, _expressionTextCache);
+            return _expressionHelper.GetExpressionText(expression);
         }
 
         protected ModelExplorer GetModelExplorer<TResult>(Expression<Func<TModel, TResult>> expression)
```

Comment:
I could make this `public` too. Would help in the `extension` scenarios without having to get `IModelExpressionProvider` from DI. Thoughts?

Reply:
I think `public` is good here but both changes require a down-cast from `IHtmlHelper<T>` and that's not great. Could we add an additional `interface` and register this class using that too?